tikv-client/src/main/java/com/pingcap/tikv/codec/MyDecimal.java

```diff
@@ -268,21 +267,22 @@ private int digitsToWords(int digits) {
    * @param size is word size which can be used in switch statement.
    * @param start start indicates the where start to read.
    */
-  private int readWord(int[] b, int size, int start) {
+  @VisibleForTesting
+  public static int readWord(int[] b, int size, int start) {
     int x = 0;
     switch (size) {
       case 1:
-        x = b[start];
+        x = (byte)b[start];
         break;
       case 2:
-        x = (b[start] << 8) + b[start + 1];
+        x = (((byte)b[start]) << 8) + (b[start + 1] & 0xFF);
         break;
       case 3:
         int sign = b[start] & 128;
         if (sign > 0) {
-          x = 255 << 24 | (b[start] & 0xFF) << 16 | (b[start + 1] & 0xFF) << 8 | (b[start + 2]);
```

Comment:
Just a reminder: b's type is byte[] which shares a range from -128 to 127 whereas golang's byte shares range from 0 to 255. We are better take `and` operation with `0xFF`.

Reply:
I don't think so, they are still the same in binary forms.

Reply:
actually, i think we should use <<<

Reply:
Additionally, `255 << 24` could be overflow. We need convert 255 to a uint and then convert back to int.

Reply:
my bad, java does not have uint..

Reply:
0xFFL << 24 | (b[start] & 0xFF) << 16 | (b[start + 1] & 0xFF) << 8 | (b[start + 2]);

Reply:
you may use ~(((b[start] & 7FFF) << 16 | (b[start + 1] & 0xFF) << 8 | (b[start + 2] & 0xFF)) + 1);

Reply:
1. b's type is int not byte[].
2. 255 as const literal is int type.

Reply:
255 << 24 has correct binary form. Numeric overflow does not affect binary. ignore my comment.